File: Domain/HistSPP.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Domain
{
    public class HistSPP : GRCBaseDomain
    {
        DateTime pOPurchnDate = DateTime.Now.Date;
        DateTime tglSPP = DateTime.Now.Date;
        DateTime dlvDate = DateTime.Now.Date;
        string noPO = string.Empty;
        string noSPP = string.Empty;
        string itemCode = string.Empty;
        string itemName = string.Empty;
        string satuan = string.Empty;
        decimal price = 0;
        string cRC = string.Empty;
        decimal qty = 0;
        string supplierName = string.Empty;
        string telepon = string.Empty;
        string termin = string.Empty;
        string delivery = string.Empty;
        decimal disc = 0;
        decimal pPN = 0;
        string approval = string.Empty;

        public DateTime DlvDate
        {
            get { return dlvDate; }
            set { dlvDate = value; }
        }
        public DateTime POPurchnDate
        {
            get { return pOPurchnDate; }
            set { pOPurchnDate = value; }
        }
        public DateTime TglSPP
        {
            get { return tglSPP; }
            set { tglSPP = value; }
        }

        public string NoPO
        {
            get { return noPO; }
            set { noPO = value; }
        }
        public string NoSPP
        {
            get { return noSPP; }
            set { noSPP = value; }
        }
        public string ItemCode
        {
            get { return itemCode; }
            set { itemCode = value; }
        }
        public string ItemName
        {
            get { return itemName; }
            set { itemName = value; }
        }
        public string Satuan
        {
            get { return satuan; }
            set { satuan = value; }
        }
        public decimal Price
        {
            get { return price; }
            set { price = value; }
        }
        public string CRC
        {
            get { return cRC; }
            set { cRC = value; }
        }
        public decimal Qty
        {
            get { return qty; }
            set { qty = value; }
        }
        public string SupplierName
        {
            get { return supplierName; }
            set { supplierName = value; }
        }
        public string Telepon
        {
            get { return telepon; }
            set { telepon = value; }
        }
        public string Termin
        {
            get { return termin; }
            set { termin = value; }
        }
        public string Delivery
        {
            get { return delivery; }
            set { delivery = value; }
        }
        public decimal Disc
        {
            get { return disc; }
            set { disc = value; }
        }
        public decimal PPN
        {
            get { return pPN; }
            set { pPN = value; }
        }
        public string Approval
        {
            get { return approval; }
            set { approval = value; }
        }


    }
}

File: GRCweb1/Scripts/Purchasing/HistSPP.js
var strField;
var strValue;

function ShowHide() {
    var x = document.getElementById("Tbldetail");
    if (x.style.display === "none") {
        x.style.display = "block";
    } else {

    }
    GetHistPO();
}

function GetHistPO() {
    strField = $("#ddlbarang").val();
    strValue = $("#txtcari").val();

    $.ajax({
        url: "HistorySPP.aspx/GetHistPO",
        type: "POST",
        contentType: "application/json; charset=utf-8",
        dataType: "json",
        data: JSON.stringify({ strField: strField, strValue: strValue }),
        success: function (data) {

            $("#tableListPO").DataTable().destroy();
            $('#tableListPO').empty();

            datatable = $.parseJSON(data.d);
            var oTblReport = $("#tableListPO");
            oTblReport.DataTable({
                "data": datatable,
                "responsive": true,
                "autoWidth": true,
                dom: 'Blfrtip',
                buttons: [
                    { extend: 'copy' },
                    { extend: 'csv' },
                    { extend: 'excel', title: 'History SPP' },
                    { extend: 'pdf', title: 'History SPP' },

                    {
                        extend: 'print', title: 'History SPP',
                        orientation: 'landscape',
                        pageSize: 'A3',
                        customize: function (win) {
                            $(win.document.body).addClass('white-bg');
                            $(win.document.body).css('font-size', '10px');

                            $(win.document.body).find('table')
                                .addClass('compact')
                                .css('font-size', 'inherit');
                        }
                    }],
                "columns": [
                    
                    {
                        "className": '',
                        "data": "TglSPP",
                        title: "Tgl SPP",
                        "render": function (data, type, row) {
                            var tgl = data;
                            var thn = tgl.substring(0, 4);
                            var bln = tgl.substring(5, 7);
                            var tgla = tgl.substring(8, 10);
                            return '<td>' + tgla + "/" + bln + "/" + thn + '</td>';
                        }
                    },
                    //{ "data": "POPurchnDate", title: "PO IssueDate" },
                    //{ "data": "DlvDate", title: "Delivery Date" },
                    { "data": "NoSPP", title: "NoSPP" },
                    {
                        "className": '',
                        "data": "POPurchnDate",
                        title: "Tgl PO",
                        "render": function (data, type, row) {
                            var tgl = data;
                            var thn = tgl.substring(0, 4);
                            var bln = tgl.substring(5, 7);
                            var tgla = tgl.substring(8, 10);
                            return '<td>' + tgla + "/" + bln + "/" + thn + '</td>';
                        }
                    },
                    //{ "data": "TglSPP", title: "TglSPP" },
                    { "data": "NoPO", title: "NoPO" },
                    //{ "data": "POPurchnDate", title: "TglPO" },
                    { "data": "ItemCode", title: "ItemCode" },
                    { "data": "ItemName", title: "ItemName" },
                    { "data": "Satuan", title: "Satuan" },
                    { "data": "Price", title: "Harga" },
                    { "data": "Qty", title: "Qty" }
                    
                ]
            });
        }
    });

    clearText();

}

function clearText()
{
    $('#txtcari').val("");
}